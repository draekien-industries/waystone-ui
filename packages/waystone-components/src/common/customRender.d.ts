/** @jsxImportSource theme-ui */
import React from 'react';
import { RenderOptions } from '@testing-library/react';
declare const DarkModeWrapper: React.FC;
declare const themedRender: (ui: React.ReactElement, options?: Omit<RenderOptions<typeof import("@testing-library/dom/types/queries"), HTMLElement, HTMLElement>, "wrapper"> | undefined) => import("@testing-library/react").RenderResult<typeof import("@testing-library/dom/types/queries"), HTMLElement, HTMLElement>;
export * from '@testing-library/react';
export { themedRender as render, DarkModeWrapper };
//# sourceMappingURL=customRender.d.ts.map