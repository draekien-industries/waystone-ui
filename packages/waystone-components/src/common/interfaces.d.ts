import { StylePropertyValue } from '@theme-ui/css';
import { Size, SizeExtended, ThemeColor, ThemeSize } from '@waystone/core';
import { Property } from 'csstype';
/** This component supports different variants. */
export interface HasVariant {
    /** The variant to render. */
    variant?: string;
}
/** This component supports different sizes. */
export interface HasSize {
    /** The size to render. */
    size?: Size;
}
/** This components supports an extended list of sizes. */
export interface HasExtendedSize {
    /** The size to render. */
    size?: SizeExtended;
}
/** This component supports a specific width. */
export interface HasWidth {
    /**
     * The width of the component.
     * Use an array to take advantage of ThemeUI breakpoints.
     */
    width?: ThemeSize | ThemeSize[] | StylePropertyValue<Property.Width<string | number>>;
}
/** This component supports a specific height. */
export interface HasHeight {
    /**
     * The height of the component.
     * Use an array to take advantage of ThemeUI breakpoints.
     */
    height?: ThemeSize | ThemeSize[] | StylePropertyValue<Property.Height<string | number>>;
}
/**
 * This component supports both a specific height and width.
 * @extends HasWidth
 * @extends HasHeight
 */
export interface HasDimensions extends HasWidth, HasHeight {
}
/** This component supports different content colors. */
export interface HasColor {
    /** The color to render the content with. */
    color?: ThemeColor | StylePropertyValue<Property.Color>;
}
/** This component supports different backgrounds. */
export interface HasBackground {
    /** The color to render as the component's background. */
    backgroundColor?: ThemeColor | StylePropertyValue<Property.BackgroundColor>;
}
/**
 * This component supports a background image.
 * @extends HasBackground
 */
export interface HasBackgroundImage extends HasBackground {
    /** The background image. */
    backgroundImage?: StylePropertyValue<Property.BackgroundImage>;
    /** The size of the background image. */
    backgroundSize?: StylePropertyValue<Property.BackgroundImage>;
}
/** This component supports extended customisation via an options record. */
export interface HasExtendedOptions {
    /** The options to apply to the component. */
    options?: Record<string, any>;
}
/** This component has a disabled state. */
export interface CanDisable {
    /** Should the component render in the disabled state? */
    disabled?: boolean;
}
/** This component has a loading state. */
export interface CanLoad {
    /** Should the component render in the loading state? */
    loading?: boolean;
}
/** This component has an activated state. */
export interface CanActivate {
    /** Should the component render in the active state? */
    active?: boolean;
}
/** This component can take up the full width of it's parent container. */
export interface CanBeFullWidth {
    /** Should the component take up the full width of it's parent container? */
    fullWidth?: boolean;
}
//# sourceMappingURL=interfaces.d.ts.map