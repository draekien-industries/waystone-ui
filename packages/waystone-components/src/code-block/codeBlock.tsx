'use client';

import {
  StyleName,
  SupportedLanguage,
  SyntaxHighlighter,
} from '@waystone/syntax-highlighter';
import { useState } from 'react';
import { Icon } from '../icon/icon';
import { copy } from './codeBlock.fx';
import {
  codeBlockCaptionCss,
  codeBlockContainerCss,
  codeBlockCopyButtonCss,
  codeBlockLanguageCss,
} from './codeBlock.styles';

export type CodeBlockProps = {
  /**
   * A flag indicating whether the code block should wrap long lines.
   * @default false
   */
  wordWrap?: boolean;
  /**
   * The text to render as code.
   */
  children: string;
  /** An optional caption to describe the code block. */
  caption?: string;

  /** The code language to use for syntax highlighting. */
  language?: SupportedLanguage;

  /** The name of the syntax highlighting style to use */
  styleName?: StyleName;
};

/**
 * Renders a code block that supports syntax highlighting using HLJS.
 * Refer to https://github.com/react-syntax-highlighter/react-syntax-highlighter/blob/HEAD/AVAILABLE_LANGUAGES_HLJS.MD
 * for the list of available languages.
 * @param props.language - the language of the code inside the codeblock. Used for syntax highlighting.
 * @param props.wordWrap - a flag indicating whether the code block should wrap long lines.
 * @param props.caption - an optional caption to describe the code block.
 * @param props.children - the code to render.
 * @returns a code block component.
 */
export const CodeBlock = ({
  language,
  wordWrap = false,
  caption,
  children,
  styleName = 'tomorrowNightBlue',
}: CodeBlockProps) => {
  const [iconName, setIconName] = useState('content_copy');

  const handleCopy = () => {
    copy(children);
    setIconName('done');
  };

  const handleBlur = () => {
    setIconName('content_copy');
  };

  return (
    <div sx={codeBlockContainerCss}>
      {language && <div sx={codeBlockLanguageCss}>{language}</div>}
      <button
        sx={codeBlockCopyButtonCss}
        type="button"
        onClick={handleCopy}
        onBlur={handleBlur}
      >
        <Icon name={iconName} />
      </button>
      <SyntaxHighlighter
        wrapLines={wordWrap}
        language={language}
        styleName={styleName}
      >
        {children}
      </SyntaxHighlighter>
      {caption && <div sx={codeBlockCaptionCss}>{caption}</div>}
    </div>
  );
};
