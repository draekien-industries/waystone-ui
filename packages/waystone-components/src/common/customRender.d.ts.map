{"version":3,"file":"customRender.d.ts","sourceRoot":"","sources":["customRender.tsx"],"names":[],"mappings":"AAAA,gCAAgC;AAChC,OAAO,KAAoB,MAAM,OAAO,CAAC;AACzC,OAAO,EAAU,aAAa,EAAE,MAAM,wBAAwB,CAAC;AAQ/D,QAAA,MAAM,eAAe,EAAE,KAAK,CAAC,EAQ5B,CAAC;AAEF,QAAA,MAAM,YAAY,OACZ,MAAM,YAAY,sQAE8B,CAAC;AAEvD,cAAc,wBAAwB,CAAC;AACvC,OAAO,EAAE,YAAY,IAAI,MAAM,EAAE,eAAe,EAAE,CAAC"}