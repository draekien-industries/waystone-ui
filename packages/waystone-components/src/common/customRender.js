import { jsx as _jsx } from "theme-ui/jsx-runtime";
/** @jsxImportSource theme-ui */
import { useEffect } from 'react';
import { render } from '@testing-library/react';
import { Waystone } from '../';
import { useColorMode } from '@theme-ui/color-modes';
const TestProvider = ({ children }) => (_jsx(Waystone, { children: children }));
const DarkModeWrapper = ({ children }) => {
    const [, setColorMode] = useColorMode();
    useEffect(() => {
        setColorMode('dark');
    }, [setColorMode]);
    return _jsx("div", { children: children });
};
const themedRender = (ui, options) => render(ui, Object.assign({ wrapper: TestProvider }, options));
export * from '@testing-library/react';
export { themedRender as render, DarkModeWrapper };
