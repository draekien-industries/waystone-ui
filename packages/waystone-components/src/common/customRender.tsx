/** @jsxImportSource theme-ui */
import React, { useEffect } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { useColorMode } from 'theme-ui';
import { Waystone } from '../waystone-ui/waystoneUi';

const TestProvider: React.FC = ({ children }) => (
  <Waystone>{children}</Waystone>
);

const DarkModeWrapper: React.FC = ({ children }) => {
  const [, setColorMode] = useColorMode();

  useEffect(() => {
    setColorMode('dark');
  }, [setColorMode]);

  return <div>{children}</div>;
};

const themedRender: (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => RenderResult = (ui, options?) =>
  render(ui, { wrapper: TestProvider, ...options });

export * from '@testing-library/react';
export { themedRender as render, DarkModeWrapper };
