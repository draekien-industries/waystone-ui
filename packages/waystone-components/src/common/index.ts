export * from './customRender';
export * from './interfaces';
