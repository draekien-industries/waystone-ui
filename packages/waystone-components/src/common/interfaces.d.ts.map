{"version":3,"file":"interfaces.d.ts","sourceRoot":"","sources":["interfaces.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,kBAAkB,EAAE,MAAM,eAAe,CAAC;AACnD,OAAO,EAAE,IAAI,EAAE,YAAY,EAAE,UAAU,EAAE,SAAS,EAAE,MAAM,gBAAgB,CAAC;AAE3E,OAAO,EAAE,QAAQ,EAAE,MAAM,SAAS,CAAC;AAEnC,kDAAkD;AAClD,MAAM,WAAW,UAAU;IACzB,6BAA6B;IAC7B,OAAO,CAAC,EAAE,MAAM,CAAC;CAClB;AAED,+CAA+C;AAC/C,MAAM,WAAW,OAAO;IACtB,0BAA0B;IAC1B,IAAI,CAAC,EAAE,IAAI,CAAC;CACb;AAED,0DAA0D;AAC1D,MAAM,WAAW,eAAe;IAC9B,0BAA0B;IAC1B,IAAI,CAAC,EAAE,YAAY,CAAC;CACrB;AAED,gDAAgD;AAChD,MAAM,WAAW,QAAQ;IACvB;;;OAGG;IACH,KAAK,CAAC,EACF,SAAS,GACT,SAAS,EAAE,GACX,kBAAkB,CAAC,QAAQ,CAAC,KAAK,CAAC,MAAM,GAAG,MAAM,CAAC,CAAC,CAAC;CACzD;AAED,iDAAiD;AACjD,MAAM,WAAW,SAAS;IACxB;;;OAGG;IACH,MAAM,CAAC,EACH,SAAS,GACT,SAAS,EAAE,GACX,kBAAkB,CAAC,QAAQ,CAAC,MAAM,CAAC,MAAM,GAAG,MAAM,CAAC,CAAC,CAAC;CAC1D;AAED;;;;GAIG;AACH,MAAM,WAAW,aAAc,SAAQ,QAAQ,EAAE,SAAS;CAAG;AAE7D,wDAAwD;AACxD,MAAM,WAAW,QAAQ;IACvB,4CAA4C;IAC5C,KAAK,CAAC,EAAE,UAAU,GAAG,kBAAkB,CAAC,QAAQ,CAAC,KAAK,CAAC,CAAC;CACzD;AAED,qDAAqD;AACrD,MAAM,WAAW,aAAa;IAC5B,yDAAyD;IACzD,eAAe,CAAC,EAAE,UAAU,GAAG,kBAAkB,CAAC,QAAQ,CAAC,eAAe,CAAC,CAAC;CAC7E;AAED;;;GAGG;AACH,MAAM,WAAW,kBAAmB,SAAQ,aAAa;IACvD,4BAA4B;IAC5B,eAAe,CAAC,EAAE,kBAAkB,CAAC,QAAQ,CAAC,eAAe,CAAC,CAAC;IAC/D,wCAAwC;IACxC,cAAc,CAAC,EAAE,kBAAkB,CAAC,QAAQ,CAAC,eAAe,CAAC,CAAC;CAC/D;AAED,4EAA4E;AAC5E,MAAM,WAAW,kBAAkB;IACjC,6CAA6C;IAC7C,OAAO,CAAC,EAAE,MAAM,CAAC,MAAM,EAAE,GAAG,CAAC,CAAC;CAC/B;AAED,2CAA2C;AAC3C,MAAM,WAAW,UAAU;IACzB,yDAAyD;IACzD,QAAQ,CAAC,EAAE,OAAO,CAAC;CACpB;AAED,0CAA0C;AAC1C,MAAM,WAAW,OAAO;IACtB,wDAAwD;IACxD,OAAO,CAAC,EAAE,OAAO,CAAC;CACnB;AAED,6CAA6C;AAC7C,MAAM,WAAW,WAAW;IAC1B,uDAAuD;IACvD,MAAM,CAAC,EAAE,OAAO,CAAC;CAClB;AAED,0EAA0E;AAC1E,MAAM,WAAW,cAAc;IAC7B,4EAA4E;IAC5E,SAAS,CAAC,EAAE,OAAO,CAAC;CACrB"}