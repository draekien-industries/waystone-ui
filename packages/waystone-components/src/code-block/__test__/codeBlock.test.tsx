import React from 'react';
import { render } from '../../common';
import { CodeBlock } from '../codeBlock';

describe('The code block component', () => {
  it('Should match snapshot', () => {
    const { container } = render(
      <CodeBlock language="javascript">
        console.Log(&quot;Hello World&quote;)
      </CodeBlock>
    );

    expect(container).toMatchSnapshot();
  });
});
