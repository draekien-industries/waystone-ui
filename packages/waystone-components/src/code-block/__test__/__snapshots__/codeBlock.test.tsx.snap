// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The code block component Should match snapshot 1`] = `
<div>
  <div
    class="css-1c9yept-CodeBlock"
  >
    <pre
      style="display: block; overflow-x: auto; background: rgb(29, 31, 33); color: rgb(197, 200, 198); padding: 0.5em; background-color: inherit; font-family: inherit; font-weight: inherit; line-height: inherit;"
    >
      <code
        class="language-javascript"
        style="white-space: pre;"
      >
        <span
          class="comment linenumber react-syntax-highlighter-line-number"
          style="display: inline-block; min-width: 1.25em; padding-right: 1em; text-align: right; user-select: none;"
        >
          1
        </span>
        <span
          style="color: rgb(222, 147, 95);"
        >
          console
        </span>
        <span>
          .Log(
        </span>
        <span
          style="color: rgb(181, 189, 104);"
        >
          "Hello World&quote;)
        </span>
      </code>
    </pre>
  </div>
</div>
`;
